flink-table/flink-table-api-java/src/main/java/org/apache/flink/table/catalog/DataTypeFactoryImpl.java

```diff
@@ -132,49 +131,12 @@ public LogicalType createLogicalType(UnresolvedIdentifier identifier) {
     private static Supplier<SerializerConfig> createSerializerConfig(
             ClassLoader classLoader, ReadableConfig config, SerializerConfig serializerConfig) {
         return () -> {
-            final SerializerConfig newSerializerConfig = new SerializerConfigImpl();
-
+            SerializerConfig newSerializerConfig = new SerializerConfigImpl();
```

Comment:
We can avoid unnecessary object creation by creating a new instance only when serializerConfig is null